appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java

```diff
@@ -471,15 +471,13 @@ protected boolean isValidTimerForThisServer(TimerPrimaryKey timerId, RuntimeTime
         return result;
     }
 
-    @Override
-    public String[] listTimers(String[] serverIds) {
+    static String [] listTimers(Collection<HZTimer> timers, String[] serverIds) {
```

Comment:
I feel like the right fix is to add generics to the `pkCache` field `IMap`. That removes the need for cast and method extraction here and in a lot of other places in the file.

Reply:
I'll prepare separate request with those changes.